mapie/tests/test_regression.py

```diff
@@ -516,46 +512,71 @@ def test_results_prefit() -> None:
 
 def test_not_enough_resamplings() -> None:
     """Test that a warning is raised if at least one residual is nan."""
-    with pytest.warns(Warning):
+    with pytest.warns(UserWarning, match=r"WARNING: at least one point of*"):
         mapie = MapieRegressor(
-            cv=JackknifeAfterBootstrap(agg_function="mean", n_resamplings=1)
+            cv=Subsample(n_resamplings=1), agg_function="mean"
         )
         mapie.fit(X, y)
 
 
-def test_invalid_agg_function() -> None:
-    """
-    Test that a wrong aggregation function in JackknifeAfterBootstrap
-    class raises an error.
-    """
-
-    with pytest.raises(ValueError, match=r".*Invalid aggregation function*"):
-        JackknifeAfterBootstrap(
-            agg_function="medium", n_resamplings=30, replace=True
+def test_no_agg_fx_specified_with_subsample() -> None:
+    """Test that a warning is raised if at least one residual is nan."""
+    with pytest.warns(Warning, match=r"WARNING: you need to specify*"):
+        mapie = MapieRegressor(
+            cv=Subsample(n_resamplings=1), agg_function=None
         )
+        mapie.fit(X, y)
 
 
 def test_invalid_randomstates() -> None:
     """
-    Test that wrong list of random states in JackknifeAfterBootstrap
+    Test that wrong list of random states in Subsample
     class raise errors.
     """
 
     with pytest.raises(
         ValueError, match=r".*Incoherent number of random states*"
     ):
-        JackknifeAfterBootstrap(
-            agg_function="mean",
+        Subsample(
             n_resamplings=30,
             replace=True,
             random_states=[0, 1],
         )
 
 
-def test_default_JackknifeAfterBootstrap() -> None:
+def test_invalid_aggregate_all() -> None:
+    """
+    Test that wrong aggregation in MAPIE raise errors.
+    """
+    with pytest.raises(
+        AssertionError,
+        match=r".*Aggregation function called but not defined.*",
+    ):
+        mapie = MapieRegressor()
+        mapie.aggregate_all(X)
+
+
+def test_invalid_aggregate_mask() -> None:
+    """
+    Test that wrong aggregation in MAPIE raise errors.
+    """
+    with pytest.raises(
+        AssertionError,
+        match=r".*Aggregation function called but not defined.*",
+    ):
+        mapie = MapieRegressor()
+        mapie.aggregate_with_mask(X, k)
+
+
+def test_default_Subsample() -> None:
     """Test default values of Jackknife+-after-Bootstrap."""
```

Comment:
"of Resample"

Reply:
A déplacer dans un module dédié (voir commentaire précédent).

Reply:
of Subsample...
Fait